cobigen-plugins/cobigen-javaplugin-parent/cobigen-javaplugin/src/test/resources/testdata/integrationtest/templates/generatedAnnotationTemplates/functions.ftl

```diff
@@ -0,0 +1,140 @@
+<#-- -------------------- -->
+<#-- devonfw SPECIFIC MACROS -->
+<#-- -------------------- -->
+
+<#--
+	Generates all field declaration whereas Entity references will be converted to appropriate id references
+-->
+<#macro generateFieldDeclarations_withRespectTo_entityObjectToIdReferenceConversion isSearchCriteria=false>
+<#list pojo.fields as field>
+<#if field.type?contains("Entity")> <#-- add ID getter & setter for Entity references only for ID references -->
+  <#if !JavaUtil.isCollection(classObject, field.name)> <#-- do not generate field for multiple relation -->
+   	 private ${field.type?replace("[^<>,]+Entity","Long","r")} ${DevonfwUtil.resolveIdVariableName(classObject,field)};
+  </#if>
+<#elseif field.type?contains("Embeddable")>
+	<#if isSearchCriteria>
+		private ${field.type?replace("Embeddable","SearchCriteriaTo")} ${field.name};
+	<#else>
+		private ${field.type?replace("Embeddable","Eto")} ${field.name};
+	</#if>
+<#elseif isSearchCriteria && JavaUtil.equalsJavaPrimitive(classObject,field.name)>
+  private ${JavaUtil.boxJavaPrimitives(classObject,field.name)} ${field.name};
+<#elseif !isSearchCriteria || !JavaUtil.isCollection(classObject, field.name)>
+	private ${field.type} ${field.name};
```

Comment:
I'm not sure if we should use this template for our tests. Currently lots of utility classes are being used here.
Also the template is way too complex for a test, needs to be reduced a lot.